crates/matrix-sdk/src/room/timeline/event_item/content.rs

```diff
@@ -143,11 +146,27 @@ impl Message {
             }
         });
 
-        Self {
-            msgtype: edit.map_or(c.msgtype, |e| e.new_content),
-            in_reply_to: c.relates_to.and_then(InReplyToDetails::from_relation),
-            edited,
-        }
+        let in_reply_to = c.relates_to.and_then(InReplyToDetails::from_relation);
+        let msgtype = edit.map_or_else(
```

Comment:
Maybe use a regular `match` instead of `map_or_else` here? Probably easier to read.